crates/driver/src/logic/eth/mod.rs

```diff
@@ -54,8 +58,82 @@ impl From<H160> for Address {
     }
 }
 
+impl From<Address> for H160 {
+    fn from(address: Address) -> Self {
+        address.0
+    }
+}
+
+/// A contract address.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct Contract(pub H160);
+
+impl From<H160> for Contract {
+    fn from(inner: H160) -> Self {
+        Self(inner.into())
+    }
+}
+
+impl From<Contract> for ethereum_types::H160 {
+    fn from(contract: Contract) -> Self {
+        contract.0 .0.into()
+    }
+}
+
+/// The contract is an address on the blockchain.
+impl From<Contract> for Address {
+    fn from(contract: Contract) -> Self {
+        contract.0.into()
+    }
+}
+
 /// An ERC20 token.
 ///
 /// https://eips.ethereum.org/EIPS/eip-20
 #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct Token(pub H160);
+
+impl From<Token> for H160 {
+    fn from(token: Token) -> Self {
+        token.0
+    }
+}
+
+/// An asset on the Ethereum blockchain. Represents a particular amount of a
+/// particular token.
+#[derive(Debug, Clone, Copy)]
+pub struct Asset {
+    pub amount: U256,
+    pub token: Token,
+}
+
+// TODO Do some web browsing to improve this doc comment.
+/// An amount of native Ether tokens.
+#[derive(Debug, Clone, Copy)]
+pub struct Ether(pub U256);
```

Comment:
Should clarify that this is measured in wei (ether atoms). Technically 1e18 wei = 1 ether.